### PR TITLE
Update libraries.mdx section for VaultSharp

### PR DESCRIPTION
**Pull request by rajanadar**
### Description

-  Added more info on VaultSharp for latest .NET version support and comprehensiveness of auth and secret backends supported
- Helps our users to choose the library easily knowing that all versions of .NET are supported, all Vault Apis are supported.

### TODO only if you're a HashiCorp employee
- [ ] **Backport Labels:** If this PR is in the ENT repo and needs to be backported, backport  
  to N, N-1, and N-2, using the `backport/ent/x.x.x+ent` labels. If this PR is in the CE repo, you should only backport to N, using the `backport/x.x.x` label, not the enterprise labels.
    - [ ] If this fixes a critical security vulnerability or [severity 1](https://www.hashicorp.com/customer-success/enterprise-support) bug, it will also need to be backported to the current [LTS versions](https://developer.hashicorp.com/vault/docs/enterprise/lts#why-is-there-a-risk-to-updating-to-a-non-lts-vault-enterprise-version) of Vault. To ensure this, use **all** available enterprise labels.
- [ ] **ENT Breakage:** If this PR either 1) removes a public function OR 2) changes the signature
  of a public function, even if that change is in a CE file, _double check_ that
  applying the patch for this PR to the ENT repo and running tests doesn't
  break any tests. Sometimes ENT only tests rely on public functions in CE
  files.
- [ ] **Jira:** If this change has an associated Jira, it's referenced either
  in the PR description, commit message, or branch name.
- [ ] **RFC:** If this change has an associated RFC, please link it in the description.
- [ ] **ENT PR:** If this change has an associated ENT PR, please link it in the
  description. Also, make sure the changelog is in this PR, _not_ in your ENT PR.

### CODE REVIEW

File: website/content/api-docs/libraries.mdx
Comment (VioletHynes) [inline website/content/api-docs/libraries.mdx:58]:
```suggestion
  - Supports .NET 4.6.x, 4.7.x, 4.8.x, .NET Standard 2.x, .NET 6.0, 7.0, and 8.0.
  - Supports every auth and secret backend supported by Vault, along with several system APIs.
```

**Review by VioletHynes — COMMENTED** [review]
Hey there! This looks good, thanks for this.

Just some minor grammatical changes before we can merge this one in, but once they're applied I'd love to.

**Review by schavis — CHANGES_REQUESTED** [review]
+1 to the changes suggested by @VioletHynes

**Comment by rajanadar** [comment]
thanks @VioletHynes  and @schavis . Done

**Review by schavis — APPROVED** [review]

**Pull request merged**
